src/indexed_crate.rs

```diff
@@ -41,6 +41,218 @@ pub struct IndexedCrate<'a> {
+                map.insert(key, value);
+                map
+            })
+            .reduce(Self::new, |l, mut r| {
+                r.merge(l);
+                r
+            })
+    }
+}
+
+impl<K: std::cmp::Eq + std::hash::Hash, V> FromIterator<(K, V)> for MapList<K, V> {
+    #[inline]
+    fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) -> Self {
+        let iter = iter.into_iter();
+        let mut map = Self::with_capacity(iter.size_hint().0);
+        for (key, value) in iter {
+            map.insert(key, value);
+        }
+        map
+    }
+}
+
+impl<K: std::cmp::Eq + std::hash::Hash, V> Extend<(K, V)> for MapList<K, V> {
+    #[inline]
+    fn extend<T: IntoIterator<Item = (K, V)>>(&mut self, iter: T) {
+        for (key, value) in iter {
+            self.insert(key, value);
+        }
+    }
+}
+
+impl<K: std::cmp::Eq + std::hash::Hash, V> MapList<K, V> {
+    #[inline]
+    #[cfg(feature = "rayon")]
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    #[inline]
+    pub fn with_capacity(capacity: usize) -> Self {
+        Self(HashMap::with_capacity(capacity))
+    }
+
+    #[inline]
+    pub fn into_inner(self) -> HashMap<K, Vec<V>> {
+        self.0
+    }
+
+    #[inline]
+    pub fn insert(&mut self, key: K, value: V) {
+        match self.0.entry(key) {
+            Entry::Occupied(mut entry) => entry.get_mut().push(value),
+            Entry::Vacant(entry) => {
+                entry.insert(vec![value]);
+            }
+        }
+    }
+
+    #[inline]
+    #[cfg(feature = "rayon")]
+    pub fn insert_many(&mut self, key: K, value: Vec<V>) {
+        match self.0.entry(key) {
+            Entry::Occupied(mut entry) => entry.get_mut().extend(value),
+            Entry::Vacant(entry) => {
+                entry.insert(value);
+            }
+        }
+    }
+
+    #[inline]
+    #[cfg(feature = "rayon")]
+    pub fn merge(&mut self, other: Self) {
+        for (key, value) in other.0 {
+            self.insert_many(key, value);
+        }
+    }
+}
+
+fn build_imports_index<'a>(
+    index: &'a HashMap<Id, Item>,
+    value: &IndexedCrate<'a>,
+) -> MapList<Path<'a>, (&'a Item, Modifiers)> {
+    #[cfg(feature = "rayon")]
+    let iter = index.par_iter();
+    #[cfg(not(feature = "rayon"))]
+    let iter = index.iter();
+
+    iter.filter_map(|(_id, item)| {
+        if !supported_item_kind(item) {
+            return None;
+        }
+        let importable_paths = value.publicly_importable_names(&item.id);
+        #[cfg(feature = "rayon")]
+        let iter = importable_paths.into_par_iter();
+        #[cfg(not(feature = "rayon"))]
+        let iter = importable_paths.into_iter();
+        Some(
+            iter.map(move |importable_path| {
+                (importable_path.path, (item, importable_path.modifiers))
+            }),
+        )
+    })
+    .flatten()
+    .collect()
+}
+
+fn build_impl_index(index: &HashMap<Id, Item>) -> MapList<ImplEntry<'_>, (&Item, &Item)> {
```

Comment:
Both of these two methods could use some doc comments that describe how they are different with and without the `rayon` feature. Those doc comments will help us create the documentation for the `rayon` feature itself, which will in turn help downstream maintainers decide whether that feature is worth enabling in their own use case.

Reply:
Fixed? Doc comments added, maybe they need more context though?